MetaMask/snaps
# feat: integrate mobile snaps execution environment and services

## **Description**
This PR adds Webview Execution Environment to `snaps-execution-environments` package. This code used to live inside mobile codebase, and was refactored to meet snaps's other EE's architecture

## **Manual testing steps**
- Clone metamask-mobile repository
- run `yarn setup`
- make sure you're on node > v18
- export `METAMASK_ENVIRONMENT=local` `and `METAMASK_BUILD_TYPE=flask`
- modify [SnapsExecutionWebView](https://github.com/MetaMask/metamask-mobile/blob/d7869a6ea6ed51b3307ac61fe57954116b2e01be/app/components/UI/SnapsExecutionWebView/SnapsExecutionWebView.tsx) adding `https://jonathansoufer.github.io` (_test only_) into envURI and replace the `uri: envURI.*` from webview accordingly.
- run `yarn start:android`
- navigate from Home to Discover/Browser.
- navigate to `https://metamask.github.io/snaps/test-snaps/latest/`
- Install any snap.

## **Screenshots/Recordings**

### **Before**
N/A

### **After**
https://github.com/MetaMask/snaps/assets/44679989/a2038a9e-1238-4696-aec8-29c8dfdace5a


## **Related issues**

This is **part** of ongoing tasks related to mobile-planning issue: 539
- https://github.com/MetaMask/mobile-planning/issues/539

## **Pre-merge reviewer checklist**


### **Snaps Message Flow Diagram**
![snaps msg flow 1 0](https://github.com/MetaMask/snaps/assets/44679989/df39f71a-9c41-4b84-b388-b88e7c43a13d)
